src/actions/LoginThunkActions.ts

```diff
@@ -1,103 +1,81 @@
+import moment from 'moment';
 import { ThunkDispatch } from 'redux-thunk';
-import { setServerConfig, ServerConfig } from '../config';
+import { setServerConfig } from '../config';
 import { HTTP_CODES } from '../types/Common';
-import { KialiAppState } from '../store/Store';
+import { KialiAppState, LoginState, Session, LoginStatus } from '../store/Store';
 import { KialiAppAction } from './KialiAppAction';
 import HelpDropdownThunkActions from './HelpDropdownThunkActions';
 import GrafanaThunkActions from './GrafanaThunkActions';
 import { LoginActions } from './LoginActions';
 import * as API from '../services/Api';
 
-const performLogin = (
-  dispatch: ThunkDispatch<KialiAppState, void, KialiAppAction>,
-  username?: string,
-  password?: string
-) => {
+import * as Login from '../services/Login';
+
+type KialiDispatch = ThunkDispatch<KialiAppState, void, KialiAppAction>;
+
+const Dispatcher = new Login.LoginDispatcher();
+
+const shouldRelogin = (state?: LoginState): boolean =>
+  !(state && state.status === LoginStatus.loggedIn && new Date() < moment(state.session!.expiresOn).toDate());
+
+const loginSuccess = async (dispatch: KialiDispatch, session: Session) => {
```

Comment:
I'm loving the `async/await`.  Like we have talked about before everything should migrate to this more concise syntax.

Reply:
Agree, next step is to make the whole api async :D